bob-tools/bin/brt.rs

```diff
@@ -250,6 +356,138 @@ fn try_main() -> AnyResult<()> {
     init_logger()?;
     log::info!("Logger initialized");
     let settings = Settings::from_matches()?;
+    match settings {
+        Settings::Recovery(settings) => recovery_command(&settings),
+        Settings::Validate(settings) => validate_command(&settings),
+        Settings::ValidateIndex(settings) => validate_index_command(&settings),
+    }
+}
+
+fn validate_index(path: &str) -> AnyResult<()> {
+    let mut reader = IndexReader::from_path(path)?;
+    let header = reader.read_header()?;
+    reader.read_filter()?;
+    for _ in 0..header.records_count {
+        if reader.is_eof() {
+            return Err(ValidationError("invalid records count, can't read more".into()).into());
+        }
+        reader.read_record_header()?;
+    }
+    Ok(())
+}
+
+fn validate_blob(path: &str) -> AnyResult<()> {
+    let mut reader = BlobReader::from_path(path)?;
+    reader.read_header()?;
+    while !reader.is_eof() {
+        reader.read_record()?;
+    }
+    Ok(())
+}
+
+fn validate_blobs_recursive(path: &str, settings: &ValidateSettings) -> AnyResult<Vec<String>> {
```

Comment:
Isn't this and validate_index_recursive have almost the same structure and differ only in leaf processing? It's better to factor out dir tree processing then